nucliadb/nucliadb/ingest/service/writer.py

```diff
@@ -909,3 +912,19 @@ def parse_model_metadata(request: KnowledgeBoxNew) -> SemanticModelMetadata:
     if request.HasField("default_min_score"):
         model.default_min_score = request.default_min_score
     return model
+
+
+def get_release_channel(request: KnowledgeBoxNew) -> utils_pb2.ReleaseChannel.ValueType:
+    """
+    Set channel to Experimental if specified in the grpc request or if the requested
+    slug has the experimental_kb feature enabled in stage environment.
+    """
+    release_channel = request.release_channel
+
+    stage_environment = running_settings.running_environment == "stage"
+    enabled_for_slug = has_feature(
+        const.Features.EXPERIMENTAL_KB, context={"slug": request.slug}
+    )
+    if stage_environment and enabled_for_slug:
```

Comment:
Move the has_feature check to the and so it is only conditionally called if we're in stage env
```suggestion
    if stage_environment and enabled_for_slug and has_feature(
        const.Features.EXPERIMENTAL_KB, context={"slug": request.slug}
    ):
```